rest/rest-sqs/src/main/scala/org/elasticmq/rest/sqs/QueueAttributesDirectives.scala

```diff
@@ -4,52 +4,94 @@ import akka.http.scaladsl.server.Route
 import org.elasticmq.rest.sqs.Action.{GetQueueAttributes, SetQueueAttributes}
 import org.elasticmq.rest.sqs.Constants._
 import org.elasticmq.rest.sqs.directives.ElasticMQDirectives
+import org.elasticmq.rest.sqs.model.RequestPayload
+import spray.json.RootJsonFormat
+import spray.json.DefaultJsonProtocol._
 
 import scala.concurrent.Future
+import scala.xml.Elem
 
 trait QueueAttributesDirectives {
-  this: ElasticMQDirectives with QueueAttributesOps =>
+  this: ElasticMQDirectives with QueueAttributesOps with AkkaSupport =>
 
-  def getQueueAttributes(p: AnyParams): Route = {
+  def getQueueAttributes(p: RequestPayload)(implicit protocol: AWSProtocol, xmlNsVersion: XmlNsVersion): Route = {
```

Comment:
why is the `xmlNsVersion` an implicit here? shouldn't it rather be a parameter for the xml request payload?

Reply:
It's used to derive other implicit typeclass: `elasticMQMarshaller`.  Maybe I could merge protocol and version into single case class and pass it as implicit?

Reply:
yes that would make sense, as the `xmlNsVersion` only applies to the xml marshaller probably?

Reply:
yes